src/main/java/esme/ui/Ui.java

```diff
@@ -0,0 +1,274 @@
+                        command + " 1')");
+            }
+            int index = generateIndex(words[1]);
+            if (!isIndexValid(index)) {
+                throw new EsmeException("Oh dear, it seems the index has wandered beyond the " +
+                        "boundaries of our list!");
+            }
+            description = taskList.deleteTask(index);
+        } catch (EsmeException e) {
+            displayLine(true);
+            System.out.println("\t" + e.getMessage());
+            displayLine(true);
+            return;
+        }
+        displayLine(true);
+        System.out.println("\t" + description + " has been removed from your destiny!");
+        displayLine(true);
+        callToWork();
+    }
+
+    public ArrayList<String> getFormattedTasks() {
+        return taskList.getFormattedTasks();
+    }
+
+    public int getNumberOfTasks() {
+        return taskList.getNumberOfTasks();
+    }
+
+    public void addTaskToList(String command, String input) {
+        String description;
+        try {
+            switch (command) {
+            case "todo":
+                description = taskList.addTodoTask(input);
+                break;
+            case "deadline":
+                description = taskList.addDeadlineTask(input);
+                break;
+            case "event":
+                description = taskList.addEventTask(input);
+                break;
+            default:
+                throw new EsmeException("Invalid command");
+            }
+        } catch (EsmeException e) {
+            displayLine(true);
+            System.out.println("\t" + e.getMessage());
+            displayLine(true);
+            return;
+        }
+        displayLine(true);
+        System.out.println("\tThe stars have aligned and " + description + " is now part of your destiny!");
+        displayLine(true);
+        callToWork();
+    }
+
+    /**
+     * Checks if the given task index is valid in terms of the task list.
+     * 
+     * @param index The index to check.
+     * @return True if the index is valid, false otherwise.
+     */
+    public boolean isIndexValid(int index) {
+        return (index <= taskList.getNumberOfTasks() && index > 0);
+    }
+
+    public boolean isTaskCompleted(int index) {
+        return taskList.getTask(index - 1).hasCompleted();
+    }
+
+    /**
+     * Marks the task at the given index as done. The index is 1-indexed since
+     * arrays are 0-indexed.
+     * 
+     * @param taskIndex The index of the task to be marked as done.
+     */
+    public void markTaskInList(int taskIndex) {
+        taskList.markTask(taskIndex - 1);
+        displayLine(true);
+        System.out.println("\tOutstanding! This task is marked as done, and your destiny shines brighter.");
+        System.out.println("\t  [X] " + taskList.getTask(taskIndex - 1).getDescription());
+        displayLine(true);
```

Comment:
You may consider extracting this part into a separate method as well as the similar pattern is used in ```unmarkTaskInList``` method as well.